### PR TITLE
vendor: update compose-go to v2.0.0-rc.3

### PR DESCRIPTION
**Pull request by crazy-max**
full diff: https://github.com/compose-spec/compose-go/compare/v1.20.0...v2.0.0-rc.3

Also revert non-deterministic compose context path changes introduced in https://github.com/docker/buildx/pull/2014 but is fixed with compose-go v2.

### CODE REVIEW

**Review by glours — APPROVED** [review]
LGTM

**Comment by crazy-max** [comment]
Context path has changed (again) and tests don't expect anymore an absolute path: https://github.com/docker/buildx/actions/runs/7640052586/job/20814449787?pr=2205#step:6:161

```
=== RUN   TestParseCompose
    compose_test.go:68: 
        	Error Trace:	/src/bake/compose_test.go:68
        	Error:      	Not equal: 
        	            	expected: "/src/bake/db"
        	            	actual  : "db"
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1 +1 @@
        	            	-/src/bake/db
        	            	+db
        	Test:       	TestParseCompose
```

This has been changed in https://github.com/docker/buildx/pull/2014 but looks like we need to revert this now with compose-go v2 since it goes to the original and proper behavior.

**Comment by crazy-max** [comment]
Unit tests don't run on Windows on CI (will fix that) but running them on a local Windows machine returns a bunch of test failures related also to context path changes in compose-go v2 which has the opposite behavior of Linux one so seems to be a bug in compose-go:

```
=== RUN   TestParseCompose
    compose_test.go:68: 
        	Error Trace:	X:/dev/www/github/docker_org/buildx/bake/compose_test.go:68
        	Error:      	Not equal: 
        	            	expected: "X:\\dev\\www\\github\\docker_org\\buildx\\bake\\db"
        	            	actual  : "db"
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1 +1 @@
        	            	-X:\dev\www\github\docker_org\buildx\bake\db
        	            	+db
        	Test:       	TestParseCompose
--- FAIL: TestParseCompose (0.00s)
```

**Comment by crazy-max** [comment]
Ok so on unix we have:

```
=== Failed
=== FAIL: bake TestParseCompose (0.01s)
    compose_test.go:68: 
        	Error Trace:	/home/runner/work/buildx/buildx/bake/compose_test.go:68
        	Error:      	Not equal: 
        	            	expected: "/home/runner/work/buildx/buildx/bake/db"
        	            	actual  : "db"
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1 +1 @@
        	            	-/home/runner/work/buildx/buildx/bake/db
        	            	+db
        	Test:       	TestParseCompose

=== FAIL: bake TestReadEmptyTargets (0.13s)
    bake_test.go:599: 
        	Error Trace:	/home/runner/work/buildx/buildx/bake/bake_test.go:599
        	Error:      	Not equal: 
        	            	expected: "/home/runner/work/buildx/buildx/bake"
        	            	actual  : "."
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1 +1 @@
        	            	-/home/runner/work/buildx/buildx/bake
        	            	+.
        	Test:       	TestReadEmptyTargets

=== FAIL: bake TestReadTargetsCompose (0.16s)
    bake_test.go:311: 
        	Error Trace:	/home/runner/work/buildx/buildx/bake/bake_test.go:311
        	Error:      	Not equal: 
        	            	expected: "/home/runner/work/buildx/buildx/bake"
        	            	actual  : "."
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1 +1 @@
        	            	-/home/runner/work/buildx/buildx/bake
        	            	+.
        	Test:       	TestReadTargetsCompose

=== FAIL: bake TestReadTargetsWithDotCompose (0.16s)
    bake_test.go:375: 
        	Error Trace:	/home/runner/work/buildx/buildx/bake/bake_test.go:375
        	Error:      	Not equal: 
        	            	expected: "/home/runner/work/buildx/buildx/bake"
        	            	actual  : "."
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1 +1 @@
        	            	-/home/runner/work/buildx/buildx/bake
        	            	+.
        	Test:       	TestReadTargetsWithDotCompose

DONE 345 tests, 1 skipped, 4 failures in 73.830s
```

on windows:

```
=== Failed
=== FAIL: bake TestParseCompose (0.01s)
    compose_test.go:68: 
        	Error Trace:	D:/a/buildx/buildx/bake/compose_test.go:68
        	Error:      	Not equal: 
        	            	expected: "D:\\a\\buildx\\buildx\\bake\\db"
        	            	actual  : "db"
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1 +1 @@
        	            	-D:\a\buildx\buildx\bake\db
        	            	+db
        	Test:       	TestParseCompose

=== FAIL: bake TestReadEmptyTargets (0.02s)
    bake_test.go:599: 
        	Error Trace:	D:/a/buildx/buildx/bake/bake_test.go:599
        	Error:      	Not equal: 
        	            	expected: "D:\\a\\buildx\\buildx\\bake"
        	            	actual  : "."
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1 +1 @@
        	            	-D:\a\buildx\buildx\bake
        	            	+.
        	Test:       	TestReadEmptyTargets

=== FAIL: bake TestReadTargetsCompose (0.11s)
    bake_test.go:311: 
        	Error Trace:	D:/a/buildx/buildx/bake/bake_test.go:311
        	Error:      	Not equal: 
        	            	expected: "D:\\a\\buildx\\buildx\\bake"
        	            	actual  : "."
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1 +1 @@
        	            	-D:\a\buildx\buildx\bake
        	            	+.
        	Test:       	TestReadTargetsCompose

=== FAIL: bake TestReadTargetsWithDotCompose (0.09s)
    bake_test.go:375: 
        	Error Trace:	D:/a/buildx/buildx/bake/bake_test.go:375
        	Error:      	Not equal: 
        	            	expected: "D:\\a\\buildx\\buildx\\bake"
        	            	actual  : "."
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1 +1 @@
        	            	-D:\a\buildx\buildx\bake
        	            	+.
        	Test:       	TestReadTargetsWithDotCompose

DONE 344 tests, 1 skipped, 4 failures in 154.523s
```

File: go.mod [hunk 1]
```diff
@@ -5,7 +5,7 @@ go 1.21
 require (
 	github.com/Masterminds/semver/v3 v3.2.1
 	github.com/aws/aws-sdk-go-v2/config v1.18.16
-	github.com/compose-spec/compose-go v1.20.0
+	github.com/compose-spec/compose-go/v2 v2.0.0-rc.2
```
Comment (glours) [inline go.mod:8]: We should wait for `compose-go/v2 v2.0.0.rc.3` release to merge this PR as we currently have regressions in this version.
There won't be any API changes just internal fixes

**Review by glours — APPROVED** [review]
LTGM, we just need to use `v2.0.0-rc.3` version of `compose-go` before merging

File: go.mod (hunk 1 above)
Comment (crazy-max) [reply go.mod:8]: Sure, let me know and I will rebase

**Review by tonistiigi — APPROVED** [review]

File: go.mod (hunk 1 above)
Comment (glours) [reply go.mod:8]: @crazy-max release `v2.0.0-rc.3` of `compose-go` is out, you can use it now

**Pull request merged**